tokens/src/lib.rs

```diff
@@ -1,21 +1,45 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use support::{decl_event, decl_module, decl_storage};
+use codec::Codec;
+use rstd::{fmt::Debug, result};
+use sr_primitives::traits::{
+	CheckedAdd, CheckedSub, MaybeSerializeDeserialize, Member, SimpleArithmetic, StaticLookup,
+};
+use srml_support::{decl_event, decl_module, decl_storage, ensure, Parameter};
+// FIXME: `srml-` prefix should be used for all srml modules, but currently `srml_system`
+// would cause compiling error in `decl_module!` and `construct_runtime!`
+// #3295 https://github.com/paritytech/substrate/issues/3295
+use srml_system::{self as system, ensure_signed};
+
 use traits::MultiCurrency;
 
-pub trait Trait: system::Trait {
-	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+pub trait Trait: srml_system::Trait {
+	type Event: From<Event<Self>> + Into<<Self as srml_system::Trait>::Event>;
+	type Balance: Parameter + Member + SimpleArithmetic + Codec + Default + Copy + MaybeSerializeDeserialize + Debug;
+	type CurrencyId: Parameter + Member + SimpleArithmetic + Default + Copy + MaybeSerializeDeserialize;
 }
 
 decl_storage! {
 	trait Store for Module<T: Trait> as Tokens {
+		/// The total issuance of a token type;
+		pub TotalIssuance get(fn total_issuance) build(|config: &GenesisConfig<T>| {
+			let issuance = config.initial_balance * (config.endowed_accounts.len() as u32).into();
+			config.tokens.iter().map(|id| (id.clone(), issuance)).collect::<Vec<_>>()
+		}): map T::CurrencyId => T::Balance;
 
+		/// The balance of a token type under an account.
+		pub Balance get(fn balance): double_map T::CurrencyId, twox_128(T::AccountId) => T::Balance;
```

Comment:
```suggestion
		pub Balance get(fn balance): double_map T::CurrencyId, blake2_256(T::AccountId) => T::Balance;
```

blake2_256 to be safe